handlers/handler_api.go

```diff
@@ -0,0 +1,111 @@
+package handlers
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// RegisterSQSHandler registers endpoints
+func (handler *SQSHandler) RegisterSQSHandler() {
+	http.HandleFunc("/server", handler.ServiceHandler)
+	http.HandleFunc("/sqs", handler.SQSHandler)
+	http.HandleFunc("/job", handler.HandleJob)
+}
+
+// ServiceHandler handles stop/start the SQS querrying service
+func (handler *SQSHandler) ServiceHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method == "PUT" {
+		val := r.URL.Query().Get("start")
+		if val == "true" {
+			if err := handler.StartServer(); err != nil {
+				fmt.Fprintf(w, fmt.Sprintf("Can not start a server. Detail %s", err))
+			} else {
+				fmt.Fprintf(w, "Successfully start a server")
+			}
+		} else if val == "false" {
+			if err := handler.ShutdownServer(); err != nil {
+				fmt.Fprintf(w, fmt.Sprintf("Can not shutdown the server. Detail %s", err))
+			} else {
+				fmt.Fprintf(w, "Successfully shutdown the server")
+			}
+		}
+
+	} else {
+		http.Error(w, "Not supported request method.", 405)
+	}
+
+}
+
+// SQSHandler handles update/get SQS URL string
+func (handler *SQSHandler) SQSHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method == "PUT" {
+		handler.QueueURL = r.URL.Query().Get("url")
+
+		if err := handler.ShutdownServer(); err != nil {
+			fmt.Fprintf(w, fmt.Sprintf("Can not shutdown the server. Detail %s", err))
+		} else {
+			fmt.Fprintf(w, "Successfully shutdown the server")
+		}
+
+		if err := handler.StartServer(); err != nil {
+			fmt.Fprintf(w, fmt.Sprintf("Can not start a server. Detail %s", err))
+		} else {
+			fmt.Fprintf(w, "Successfully start a server")
+		}
+
+	} else if r.Method == "GET" {
+		fmt.Fprintf(w, handler.QueueURL)
+	} else {
+		http.Error(w, "Not supported request method.", 405)
+	}
+}
+
+func (handler *SQSHandler) HandleJob(w http.ResponseWriter, r *http.Request) {
+	if r.Method == "GET" {
+		handler.listJob(w, r)
+	} else if r.Method == "POST" {
+		handler.addJob(w, r)
```

Comment:
so this endpoint is for jobconfig CRUD, not job CRUD ? if so, the function names are quite confusing
also I couldn't get a clean idea of the whole logic flow by looking at the code